### PR TITLE
Add new file_content_type for PreviewProvider subclasses

### LINKED ISSUE #2860: file_types_order: false positive with PreviewProvider subclasses in SwiftUI
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`file_types_order` triggers when multiple classes are declared in the same file:

```swift
class Foo {
}

class Bar: Foo {
}
```

### Environment

* SwiftLint version: `0.35.0`
* Installation method used: `Homebrew`

### PR DESCRIPTION
**Pull request by Jeehut**
Fixes #2860.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Style/FileTypesOrderRuleExamples.swift [hunk 1]
```diff
@@ -130,6 +118,17 @@ internal struct FileTypesOrderRuleExamples {
         extension Foo {}
         extension Bar {
         }
+        """,
+        """
```
Comment (ZevEisenberg) [inline Source/SwiftLintFramework/Rules/Style/FileTypesOrderRuleExamples.swift]: Due to #3040, please wrap all new examples in `Example(...)`.

Comment (Jeehut) [reply Source/SwiftLintFramework/Rules/Style/FileTypesOrderRuleExamples.swift]: Thanks for the pointer, fixed via rebase to solve merge conflicts (where I also fixed this).

**Comment by PSchmiedmayer** [comment]
@Jeehut I think this is a great addition to the file_content_type rule, thanks! 
@PaulTaykalo @ZevEisenberg Any update to when it would be possible to merge this into master? Any way I could help there?

**Comment by MortyMerr** [comment]
@Jeehut @PaulTaykalo @ZevEisenberg this feature is mentioned in 0.39.0 release notes in CHANGELOG, however, it's not available in pod, which is very confusing and looks like an accidental mistake. Is there anything blocking that from release? It would be awesome to get this feature :)
Sincerely

**Pull request merged**